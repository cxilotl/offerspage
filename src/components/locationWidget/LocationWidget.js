import React from 'react';
import cssStyles from './LocationWidget.module.scss';

const LocationWidget = () => {
  const label = 'Location';
  const selectedLocation = 'Fitzrovia';
  const SelectLocationLabel = 'Change Location';
  const onSelectLocation = () => {
    console.log('Selecting New Location');
  };
  return(
    <>
      <div className={ cssStyles.mainInfo }>
        <span className={ cssStyles.label }>{ label }</span>
        <span className={ cssStyles.selectedLocation }>{ selectedLocation }</span>
      </div>
      <button id="locationUpdate" name="locationUpdate"
              data-testid="location-button" className={ cssStyles.locationSelectionBtn }
              type="button" onClick={ onSelectLocation }>
        { SelectLocationLabel }
      </button>
    </>
  );
};

export default LocationWidget;