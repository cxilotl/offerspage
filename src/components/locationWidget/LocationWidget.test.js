import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import LocationWidget from './LocationWidget';

describe('LocationWidget', () => {

  it('Snapshot', () => {
    const component = renderer.create(
      <LocationWidget />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Given an LocationWidget element, it', () => {
    afterEach(cleanup);

    it('should include a label', () => {
      const { getByText } = render(
        <LocationWidget />
      );
      const label = getByText('Location');
      expect(label).toBeInTheDocument();
    });

    it('should include a selected location', () => {
      const { getByText } = render(
        <LocationWidget />
      );
      const selectedLocation = getByText(/Fitzrovia/i);
      expect(selectedLocation).toBeInTheDocument();
    });

    it('should handle a button for selecting a new location', () => {
      const { getByTestId } = render(
        <LocationWidget />
      );
      const locationButton = getByTestId('location-button');
      expect(locationButton).toBeInTheDocument();
      expect(locationButton.innerHTML).toBe('Change Location');
    });
  });

});