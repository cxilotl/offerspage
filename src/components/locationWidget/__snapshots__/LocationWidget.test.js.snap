// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationWidget Snapshot 1`] = `
Array [
  <div
    className="mainInfo"
  >
    <span
      className="label"
    >
      Location
    </span>
    <span
      className="selectedLocation"
    >
      Fritzovia
    </span>
  </div>,
  <button
    className="locationSelectionBtn"
    data-testid="location-button"
    id="locationUpdate"
    name="locationUpdate"
    onClick={[Function]}
    type="button"
  >
    Change Location
  </button>,
]
`;
